tekton/pipeline_run.go

```diff
@@ -135,6 +138,34 @@ func (r *ReleasePipelineRun) WithEnterpriseContractPolicy(enterpriseContractPoli
 	return r
 }
 
+// copyPipelinesAsCodeMeta copies key/value pairs in src adding them to dst.
+// When a key in src is already present in dst, no copy happens.
+// When the src key does not contain the prefix string, no copy happens.
+func copyPipelinesAsCodeMeta(src, dst map[string]string) {
```

Comment:
We can keep this within the function as it's not used by any other function here. If that happens in the future, we will extract it to a more generic package.

Reply:
Giving this a second thought, you use it twice in the same function... we should probably extract it already. Thinking on the best way to do this package naming wise.

Reply:
We could have a `metadata` package with a `filter.go` file. Inside that file we would have two exported functions:
- `GetAnnotationsWithPrefix(object client.Object, prefix string)`
- `GetLabelsWithPrefix(object client.Object, prefix string)`

You could still have an utility (non-exported) function in this new file for shared code.

What do you think @johnbieren?

Reply:
Good idea. I created a separate `metadata` package, and exposed an interface containing `GetAnnotationsWithPrefix` and `GetLabelsWithPrefix`, and two other exported methods to Copy Annotations & Labels metadata. The interface is mocked in `pipeline_run_test` so not to introduce any external effects on existing code, and fully unit tested & covereed in it's own package. PTAL.